### PR TITLE
feat: Support ECDSA Secp256k1 linked data signature suite

### LINKED ISSUE #1529: Support ECDSA Secp256k1 linked data signature suite
https://w3c-ccg.github.io/ld-cryptosuite-registry/#ecdsa-secp256k1

https://w3c-dvcg.github.io/lds-ecdsa-secp256k1-2019/

### PR DESCRIPTION
**Pull request by kdimak**
closes #1529

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

### CODE REVIEW

**Comment by kdimak** [comment]
`ECDSA Secp256k1 linked data signature` will be a third linked data suite supported. There are some common parts that can be extracted into separate functions/structures. I will see how to refactor this in the follow-up PR.

File: pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	if pubKey.Type != jwkType {
+		return ErrTypeNotEcdsaSecp256k1VerificationKey2019
+	}
+
+	if pubKey.JWK.Kty != "EC" {
+		return fmt.Errorf("unsupported key type: '%s'", pubKey.JWK.Kty)
+	}
+
+	if pubKey.JWK.Crv != "" && pubKey.JWK.Crv != "secp256k1" {
```
Comment (Baha-sk) [inline pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go:84]: `pubKey.JWK.Crv != ""` is not needed, Crv must be `secp256k1` regardless

File: pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		return fmt.Errorf("ecdsa: not secp256k1 curve: '%s'", pubKey.JWK.Crv)
+	}
+
+	if pubKey.JWK.Algorithm != "" && pubKey.JWK.Algorithm != "ES256K" {
```
Comment (Baha-sk) [inline pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go:88]: same for Algorithm

File: pkg/doc/verifiable/embedded_proof.go [hunk 3]
```diff
@@ -18,7 +18,9 @@ func mustBeLinkedDataProof(proofMap map[string]interface{}) error {
 	}
 
 	proofTypeStr := safeStringValue(proofType)
-	if proofTypeStr != "Ed25519Signature2018" && proofTypeStr != "JsonWebSignature2020" {
+	if proofTypeStr != "Ed25519Signature2018" &&
```
Comment (Baha-sk) [inline pkg/doc/verifiable/embedded_proof.go:21]: can be replaced by a switch case like:
```
switch proofTypeStr {
case "Ed25519Signature2018", "JsonWebSignature2020", "EcdsaSecp256k1Signature2019":
   return nil
default:
   return fmt.Errorf("unsupported proof type: %s", proofType)
}
```

File: pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go:84]: `crv` and `alg` are not mandatory fields of JSON Web Key (https://tools.ietf.org/html/rfc7517). That's why I put a soft check here.  For example,  at [reference impl](https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js) they use `kty` and `crv` only: https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js/blob/461bc062107b262e15c945b4fd02ca2e1ab22c27/packages/lds-ecdsa-secp256k1-2019-demo/src/__tests__/did-elem.spec.js#L18

Comment (Baha-sk) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go:84]: ok

Comment (kdimak) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go:84]: OK, I've re-checked it, `crv` is mandatory for Elliptic Curve Keys, I will improve that: https://tools.ietf.org/html/draft-ietf-jose-json-web-key-00#section-4.2.1

Comment (kdimak) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go]: Fixed, thank you @Baha-sk for the review comment!

Comment (Baha-sk) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go]: I think Algorithm is also required to specify the bitcoin alg below, you can remove the `Algorithm != ""` condition from there too to be consistent with the other checks in this function.

Comment (kdimak) [reply pkg/doc/signature/suite/ecdsasecp256k1signature2019/public_key_verifier.go]: I conceptually agree with you. But we want to be compatible with other implementations, e.g. https://github.com/decentralized-identity/lds-ecdsa-secp256k1-2019.js/blob/461bc062107b262e15c945b4fd02ca2e1ab22c27/packages/lds-ecdsa-secp256k1-2019-demo/src/__tests__/did-elem.spec.js#L18. Also, `alg` is not mandatory at https://tools.ietf.org/html/rfc7517. So let's relax it.

**Pull request merged**